// Copyright 2016 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// -----------------------------------------------------------------------------

syntax = "proto3";

option java_multiple_files = true;
option java_package = "sawtooth.sdk.protobuf";

message TransactionHeader {
    string batcher = 1;
    repeated string dependencies = 2;
    string family_name = 3;
    string family_version = 4;
    repeated string inputs = 5;
    string nonce = 6;
    repeated string outputs = 7;
    string payload_encoding = 8;
    string payload_sha512 = 9;
    string signer = 10;
}

message Transaction {
    bytes header = 1;
    bytes payload = 2;
    string signature = 3;
}
